frontend/src/components/Collapse/index.tsx

```diff
@@ -0,0 +1,44 @@
+import clsx from "clsx";
+import { CSSProperties, JSX, PropsWithoutRef, useEffect, useRef, useState } from "react";
+import styles from "./Collapse.module.css";
+
+/* This value should match with transition duration value */
+const TRANSITION_DURATION = 350;
+
+type CollapseProps = Overwrite<PropsWithoutRef<JSX.IntrinsicElements["div"]>, {
+  open: boolean;
+}>;
+
+export default function Collapse({ children, className, open, ...props }: CollapseProps) {
+  const contentRef = useRef(null);
+  const [contentHeight, setContentHeight] = useState<number>(0);
+  const [overflow, setOverflow] = useState<CSSProperties["overflow"]>("hidden");
```

Comment:
I don't have much against this, but I feel a single `state` is better. It is not really applicable for this case, but updating multiple states in one place might cause weird re-rendering behaviors and effect triggering. But then, looks like they did some kind of "magic" to prevent this from happening: https://react.dev/blog/2022/03/29/react-v18#new-feature-automatic-batching

Not sure 🤷‍♂️